public_html/wp-content/plugins/pattern-directory/includes/favorite.php

```diff
@@ -84,3 +84,81 @@ function get_favorites( $user = 0 ) {
 
 	return array_map( 'absint', $favorites );
 }
+
+/**
+ * Get the cached count of how many times this pattern has been favorited.
+ *
+ * @param int|WP_Post $post The pattern to check.
+ * @return integer
+ */
+function get_favorite_count( $post = 0 ) {
+	$post = get_block_pattern( $post );
+	if ( ! $post ) {
+		return false;
+	}
+
+	return absint( get_post_meta( $post->ID, META_KEY, true ) );
+}
+
+/**
+ * Get a count of how many times this pattern has been favorited, directly from the users table.
+ *
+ * @param int|WP_Post $post The pattern to check.
+ * @return integer
+ */
+function get_raw_favorite_count( $post = 0 ) {
+	global $wpdb;
+	$post = get_block_pattern( $post );
+	if ( ! $post ) {
+		return false;
+	}
+	$count = $wpdb->get_var( $wpdb->prepare(
+		"SELECT COUNT(*)
+			FROM {$wpdb->usermeta}
+			WHERE meta_key=%s
+			AND meta_value=%d",
+		META_KEY,
+		$post->ID
+	) );
+
+	return absint( $count );
+}
+
+/**
+ * Update a given post's favorite count cache.
+ *
+ * @param mixed $value The post ID.
```

Comment:
For clarity and consistency, it might be better for this function's parameter to be named `$post_id` and set its type to `int` in the doc block.

Reply:
I updated this to be `post_id`, but the value _could_ technically be anything, so I wouldn't want something to go wrong if someone expected it to be an int.